api/src/main/java/org/openmrs/api/db/hibernate/HibernateAdministrationDAO.java

```diff
@@ -203,18 +211,82 @@ public void setApplicationContext(ApplicationContext applicationContext) throws
 	 */
 	@Override
 	public void validate(Object object, Errors errors) throws DAOException {
+		
+		Class entityClass = object.getClass();
+
+		ClassMetadata metadata = sessionFactory.getClassMetadata(entityClass);
+		if (metadata != null) {
+			String[] propNames = metadata.getPropertyNames();
+			Object IdentifierTpye = metadata.getIdentifierType();
+			String identifierName = metadata.getIdentifierPropertyName();
+			if (IdentifierTpye instanceof StringType && identifierName != null) {
+				ValidateUtil.validateFieldLengths(errors, object.getClass(), identifierName);
+			}
+			for (int i = 0; i < propNames.length; i++) {
+				String propType = metadata.getPropertyType(propNames[i]).getName();
+				if (propType.equalsIgnoreCase("String")) {
```

Comment:
you don't need to check for java.lang.String or does hibernate automatically convert it to string.
You also need to check for text